examples/rvc-app/rvc-common/src/rvc-device.cpp

```diff
@@ -0,0 +1,107 @@
+#include "rvc-device.h"
+
+using namespace chip::app::Clusters;
+
+void RvcDevice::Init()
+{
+    mRunModeInstance.Init();
+    mCleanModeInstance.Init();
+    mOperationalStateInstance.Init();
+}
+
+void RvcDevice::HandleRvcRunChangeToMode(uint8_t NewMode, ModeBase::Commands::ChangeToModeResponse::Type & response)
+{
+    uint8_t currentMode = mRunModeInstance.GetCurrentMode();
+
+    switch (NewMode)
+    {
+    case RvcRunMode::ModeMapping: {
+        // change to mapping only allowed from the Idle state.
+        if (currentMode != RvcRunMode::ModeIdle)
+        {
+            response.status = to_underlying(ModeBase::StatusCode::kGenericFailure);
+            response.statusText.SetValue(chip::CharSpan::fromCharString("Change to the mapping mode is only allowed from idle"));
+            return;
+        }
+
+        mRunModeInstance.UpdateCurrentMode(NewMode);
+        // Set operational state to running
+        ChipError err =
+            mOperationalStateInstance.SetOperationalState(to_underlying(OperationalState::OperationalStateEnum::kRunning));
+        if (err != CHIP_NO_ERROR)
+        {
+            response.status = to_underlying(ModeBase::StatusCode::kGenericFailure);
+            response.statusText.SetValue(chip::CharSpan::fromCharString(err.AsString()));
```

Comment:
StatusText should always be user readable or empty. Is that also true for err.AsString()?

Reply:
I should be. I guess it depends who the user is. But this is an example. People using this example can choose to put what they want.